Assets/Vimeo/Scripts/Services/VimeoUploader.cs

```diff
@@ -0,0 +1,176 @@
+using System.IO;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Networking;
+using SimpleJSON;
+using Vimeo;
+
+namespace Vimeo
+{
+    class VimeoUploader : MonoBehaviour
+    {
+        //Events
+        public delegate void UploadAction(string status, float progress = 0.0f);
+        public event UploadAction OnUploadProgress;
+        public event UploadAction OnUploadError;
+        public event UploadAction OnUploadComplete;
+
+        public delegate void UploadEvent(VideoChunk chunk, string msg = "");
+        public event UploadEvent OnChunckUploadComplete;
+        public event UploadEvent OnChunckUploadError;
+
+        //Private members
+        private Queue<VideoChunk> myChunks;
+        public VimeoApi vimeoApi;
+        //TODO: In the future this will be stored in a hash table to provide batch uploading
+        private string file;
+        private string vimeo_url;
+        private FileInfo fileInfo;
+
+
+        //Public members
+        public int maxChunkSize;
+        public int numChunks;
+        private void Start()
+        {
+            this.hideFlags = HideFlags.HideInInspector;
+        }
+
+        public void Init(string _token, int _maxChunkSize = 1000)
+        {
+            //A queue of video chunks to upload
+            myChunks = new Queue<VideoChunk>();
+
+            //Instantiate the Vimeo Api
+            if (vimeoApi == null) {
+                vimeoApi = gameObject.AddComponent<VimeoApi>();
+                vimeoApi.OnError += ApiError;
+                vimeoApi.OnNetworkError += NetworkError;
+                vimeoApi.OnRequestComplete += RequestComplete;
+
+                vimeoApi.token = _token;
+            }
+
+            maxChunkSize = _maxChunkSize;
+
+        }
+        private void RequestComplete(string response)
+        {
+            string tusUploadLink = GetTusUploadLink(response);
+            vimeo_url = GetVideoPermlink(response);
+            CreateChunks(file, fileInfo, tusUploadLink);
+
+            //Kick off first chunks, others will be called on OnCompleteChunk()
+            VideoChunk firstChunk = myChunks.Dequeue();
+            firstChunk.Upload();
+        }
+        public void SetChunkSize(int size)
+        {
+            maxChunkSize = size;
+        }
+        private void ApiError(string response)
+        {
+
```

Comment:
Complete these methods

Reply:
Instead of re-implementing everything in the `VimeoUploader` it now extends `VimeoApi` so we don't need to worry about re-surfacing methods - https://github.com/vimeo/vimeo-unity-sdk/pull/46/commits/2427d90deb58cbaf846728cff0a46aec75b7e04b